activesupport/lib/active_support/message_verifier.rb

```diff
@@ -74,8 +77,8 @@ def valid_message?(signed_message)
     def verified(signed_message)
       if valid_message?(signed_message)
         begin
-          data = signed_message.split("--")[0]
-          @serializer.load(decode(data))
+          data = signed_message.split(".")[1]
+          @serializer.load(decode(data)[0]).symbolize_keys
```

Comment:
Ah okay, I see you're symbolizing keys. It might be better to use `with_indifferent_access` here.